### ISSUE TITLE
RequestServer should allow for externalization of realpath method, allowing for better current working directory support.

### ISSUE THREAD

**Issue by robermorales** [pkg/sftp#230]
There is a minimal issue in the current implementation of `cleanPath` in `request-server.go@223`:

```golang
// Makes sure we have a clean POSIX (/) absolute path to work with
func cleanPath(p string) string {
	p = filepath.ToSlash(p)
	if !filepath.IsAbs(p) {
		p = "/" + p
	}
	return path.Clean(p)
}
```

It does the assumption of current working directory to be "/", since "/" got prepended when path is relative.

SFTP RFC supports the idea of remote working directory, and most clients send a request to clarify which it is, just after connecting. Actually, the client included in this package also provides the functionality of `Getwd` and implement it sending a request of `SSH_FXP_REALPATH` with the path `"."`

The assumption made in `cleanPath` is clearly not what most of servers implement it, so we are forcing every RequestServer to start at "/", without providing a way to change it.

Here, the same file `request-server.go`, line `165`:

```golang
		case *sshFxpRealpathPacket:
			rpkt = cleanPacketPath(pkt)
		case *sshFxpOpendirPacket:
			request := requestFromPacket(ctx, pkt)
			handle := rs.nextRequest(request)
			rpkt = sshFxpHandlePacket{pkt.id(), handle}
```

We can see that the request for the realpath is not proxied to the RequestServer `rs`.

I propose redirect this request to the `rs`, to the `Filelist` handler, and allow the creator of every RequestServer to specify how to manage this.

If the issue got some attention and seems reasonable to be accepted, I can implement it and send a PR, with tests, modifying the documentation present at `request-readme.md` and the example at `request-example.go`.

Current implementation proposal will not break current 3rd party code, since we are not adding more methods to the interface FileLister.

Related issues refer to the ability to expose this option in the file based server, but the current issue is related to the RequestServer, to allow the functionality to be implemented outside, in the handlers.

https://github.com/pkg/sftp/issues/82
https://github.com/pkg/sftp/pull/92

So, should I implement the behaviour and send the PR?

**Comment by eikenb** [comment]
Thanks for contributing!

The current code does not allow for an starting remote working directory as a way simplify things. Now that the code has mostly stabilized I am more than willing to entertain adding it. I'd need to see the code before saying for sure it'd be accepted, but I'm willing to review and (possibly) merge a PR.

TLDR; Thanks and please submit the PR.
